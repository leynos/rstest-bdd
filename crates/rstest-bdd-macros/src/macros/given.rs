//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
