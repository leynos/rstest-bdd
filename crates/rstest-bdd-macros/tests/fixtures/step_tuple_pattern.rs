use rstest_bdd_macros::given;

#[given("coordinates")]
fn step_with_tuple((x, y): (i32, i32)) {}

fn main() {}
