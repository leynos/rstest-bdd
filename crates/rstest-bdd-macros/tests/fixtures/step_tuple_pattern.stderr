error: complex destructuring patterns are not yet supported - pattern resolves to 2 identifiers but a single bare identifier is required
 --> tests/fixtures/step_tuple_pattern.rs:4:20
  |
4 | fn step_with_tuple((x, y): (i32, i32)) {}
  |                    ^^^^^^
